lib/cn/csched_sp3.c

```diff
@@ -1036,18 +1029,27 @@ sp3_dirty_node_locked(struct sp3 *sp, struct cn_tree_node *tn)
                 sp3_node_remove(sp, spn, RBT_L_PCAP);
             }
         }
+    } else {
+
+        /* If this root node is ready to spill then ensure it's on the list
+         * in FIFO order, retaining its current position if it's already on
+         * the list.  List order is otherwise managed by sp3_check_roots().
+         */
+        if (nkvsets >= sp->thresh.rspill_kvsets_min && jobs < 3) {
```

Comment:
Where do the jobs limits of 1 and 3 come from?

Reply:
We can never run more than one job at a time on a leaf node.
We have at most 5 threads available for root node processing, but we choose to not let more than 3 of those work on any given root node.  I could probably make a #define for that..

Reply:
Added a comment instead.

Reply:
Yea #defines would be great here with some comments!